src/fund_view_module.rs

```diff
@@ -0,0 +1,231 @@
+                        if let FundType::Free{ requested_unstake: true } = fund_info.fund_type { true } else { false })
+            },
+            UserStakeState::ActivationFailed => {
+                self.fund_module().query_list(DISCR_ACTIVE_FAILED, 
+                    |fund_info|
+                        (user_id == USER_STAKE_TOTALS_ID || fund_info.user_id == user_id))
+            },
+            UserStakeState::ActiveForSale => {
+                self.fund_module().query_list(DISCR_ACTIVE_FOR_SALE, 
+                    |fund_info|
+                        (user_id == USER_STAKE_TOTALS_ID || fund_info.user_id == user_id))
+            },
+            UserStakeState::PendingDeactivationFromSale => {
+                self.fund_module().query_list(DISCR_PENDING_DEACT,
+                    |fund_info|
+                        (user_id == USER_STAKE_TOTALS_ID || fund_info.user_id == user_id) &&
+                        if let FundType::PendingDeactivation{ requested_unstake: true } = fund_info.fund_type { true } else { false }
+                )
+            },
+        }
+    }
+
+
+    fn get_user_total_stake(&self, user_id: usize) -> BigUint {
+        self.fund_module().query_all(
+            |fund_info| fund_info.fund_type.is_stake() && fund_info.user_id == user_id)
+    }
+
+    /// Yields how much a user has staked in the contract.
+    #[view(getUserStake)]
+    fn get_user_total_stake_endpoint(&self, user_address: Address) -> BigUint {
+        let user_id = self.user_data().get_user_id(&user_address);
+        if user_id == 0 {
+            BigUint::zero()
+        } else {
+            self.get_user_total_stake(user_id)
+        }
+    }
+
+    #[view(getUserActiveStake)]
+    fn get_user_active_stake_endpoint(&self, user_address: Address) -> BigUint {
+        let user_id = self.user_data().get_user_id(&user_address);
+        if user_id == 0 {
+            BigUint::zero()
+        } else {
+            self.get_user_stake_of_type(user_id, UserStakeState::Active)
+        }
+    }
+
+    #[view(getUserInactiveStake)]
+    fn get_user_inactive_stake_endpoint(&self, user_address: Address) -> BigUint {
+        let user_id = self.user_data().get_user_id(&user_address);
+        if user_id == 0 {
+            BigUint::zero()
+        } else {
+            self.get_user_stake_of_type(user_id, UserStakeState::Inactive) +
+            self.get_user_stake_of_type(user_id, UserStakeState::WithdrawOnly)
+        }
+    }
+
+    fn get_user_stake_by_type(&self, user_id: usize) -> MultiResult10<BigUint, BigUint, BigUint, BigUint, BigUint, BigUint, BigUint, BigUint, BigUint, BigUint> {
+        (
+            self.get_user_stake_of_type(user_id, UserStakeState::Inactive),
+            self.get_user_stake_of_type(user_id, UserStakeState::PendingActivation),
+            self.get_user_stake_of_type(user_id, UserStakeState::Active),
+            self.get_user_stake_of_type(user_id, UserStakeState::PendingDeactivation),
```

Comment:
the number of states can become smaller. Inactive does not need a list. Wating, Staked, UnStaked. From unStaked it can go directly to inActive - if conditions fullfilled.